### DISCUSSION TITLE
At runtime, change the database connection in auth.authenticate()

### DISCUSSION THREAD

**Discussion by mkevison** [adonisjs/core#4088] [Show and tell]
If you have a multi-tenant with each tenant having its own database this is required

I didn't find it in the documentation but thanks to the typescript I was able to do it

All you need to do is
`
auth.use('your-guard').config.provider.connection = connection`

where yout-guard is your guard defined in config/auth.ts

And connection is the name of your database connection

after that you can call authenticate() method

**Comment by adam-lages** [comment]
Good night Miquéias, how are you?
Can you please provide your explanation in more detail? I don't have much knowledge in mult-tenant and I would like to understand a little better.

Thank you very much.

**Reply by mkevison** [reply]
I have a main database and I have a database for each tenant, if I have two tenants I will have a total of 3 databases, the main one and 1 for each tenant.

I will give a more simplified example because my use is a little more complex than the example I will give below.

I created a table in my main database called tenants.

In this table I have the following information.

```
id -> unique id for each tenant
host -> url of the tenant where I identify it by the request url
connection -> the name of the connection to my database
name -> tenant name.
```

In all requests to my api I send the following information in the header

`{ tenant: request_url }`

In my api I use `request.header(tenant)` and through this url sent by the header I identify the tenant by my table in the database.

Based on that I get my tenant's connection to know which database I should work on and save the tenant's connection in cookies so I don't have to double-check the database on every request.
https://docs.adonisjs.com/guides/cookies

With the connection I can send the queries to my tenant's database that is requesting it.

So as I have a database for each tenant I need to know the connection name so I can auth.attempt()

I have a single Front end for all my tenants
if the site is loaded by the **tenant-example.com** domain, it will load its own favicon, its own colors, its own products, etc., all this without doing any configuration in the front end, just getting the information in the api.

I also created a middleware that checks if the tenant is being sent in the header.

I know that most browsers already send the url request in the header, but I chose to do it that way, since I don't know in which browser types the client will access the site.

If you have any further questions I'd be happy to help.
